packages/types/src/auth.ts

```diff
@@ -25,3 +25,42 @@ export interface AuthScheme {
   signingScope?: never;
   properties: Record<string, unknown>;
 }
+
+// As described in the Smithy documentation:
+// https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy
+export interface HttpHeaderAuthDefinition {
+  /**
+   * The Auth value will be transported in the named header,
+   * optionally prefixed with the provided scheme.
+   */
+  in: HttpAuthLocation.HEADER;
+
+  /**
+   * The name of the header parameter to use for the transporting the Auth value.
+   */
+  name: string;
+
+  /**
+   * The scheme to use.
+   */
+  scheme?: string;
+}
+
+export interface HttpQueryAuthDefinition {
+  /**
+   * The Auth value will be transported in the named query parameter.
+   */
+  in: HttpAuthLocation.QUERY;
+
+  /**
+   * The name of the query parameter to use for the transporting the API key.
+   */
+  name: string;
+}
+
+export enum HttpAuthLocation {
+  HEADER = "header",
+  QUERY = "query",
+}
+
+export type HttpAuthDefinition = HttpHeaderAuthDefinition | HttpQueryAuthDefinition;
```

Comment:
Redundant interfaces can be removed
```suggestion
export interface HttpAuthDefinition {
  /**
   * Defines the location of where the key is serialized.
   */
  in: HttpApiKeyLocation;

  /**
   * Defines the name of the HTTP header or query string parameter
   * that contains the API key.
   */
  name: string;

  /**
   * Defines the security scheme to use on the `Authorization` header value.
   * This can only be set if the "in" property is set to {@link HttpApiKeyLocation.HEADER}.
   */
  scheme?: string;
}

export enum HttpApiKeyLocation {
  HEADER = "header",
  QUERY = "query",
}
```